typing/typing_recovery.mli

```diff
@@ -0,0 +1,51 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                Ulysse Gérard, Thomas Refis, Frederic Bour              *)
+(*                           Xavier Van de Woestyne                       *)
+(*                                                                        *)
+(*   Copyright 2025 Tarides                                               *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+exception Resume
+
+(** Returns a reference initially set to false that will be set to true
+    when a type error is raised. *)
+val monitor_errors : unit -> bool ref
+
+(** Raise an error that can be caught: normal flow is resumed if a
+    [catch_errors] handler was installed. *)
+val raise_error : exn -> unit
+
+(** Any [raise_error] invoked inside catch_errors will be added to
+    the list. *)
+val catch_errors : Warnings.state -> exn list ref -> (unit -> 'a) -> 'a
+
+(** Re-raise an exception*)
+val resume_raise : exn -> 'a
+
+(* Keep track of type variables generated by error recovery. *)
+
+val erroneous_type_register : Types.type_expr -> unit
+val erroneous_type_check : Types.type_expr -> bool
+val erroneous_expr_check : Typedtree.expression -> bool
+
+
+(** Extend the given attributes with an incorrect attribute and the
+    saved types after turning them into attributes *)
+val recovery_attributes : Parsetree.attributes -> Parsetree.attributes
+
+(** Turn saved types from Cmt_format into attributes *)
+val flush_saved_types : unit -> Parsetree.attributes
```

Comment:
This is already called from `recovery_attributes`, does it need to be exposed? Or, rather: is it OK to call this function instead of `recovery_attributes`?